Using framework on OpenShift 4

### Describe the problem:

  * On OpenShift 4, the framework does not recognize OpenShift:
2019-04-18 19:33:39 INFO  Entrypoint:217 - https://172.30.0.1/oapi returned 404. We are not on OpenShift. Assuming, we are on Kubernetes. As a result, Kubernetes not OS client is used and everything fails

#### Steps to reproduce:

  1. Try to run operator on OpenShift 4
  
#### Observed Results:

 VM settings:
    Max. Heap Size (Estimated): 3.76G
    Ergonomics Machine Class: server
    Using VM: OpenJDK 64-Bit Server VM

2019-04-18 19:33:38 INFO  Entrypoint:56 - Starting..
2019-04-18 19:33:39 INFO  Entrypoint:217 - https://172.30.0.1/oapi returned 404. We are not on OpenShift. Assuming, we are on Kubernetes.
2019-04-18 19:33:39 INFO  Manifests:193 - 55 attributes loaded from 52 stream(s) in 51ms, 55 saved, 724 ignored: ["Ant-Version", "Archiver-Version", "Automatic-Module-Name", "Bnd-LastModified", "Build-Jdk", "Build-Timestamp", "Built-By", "Bundle-ClassPath", "Bundle-ContactAddress", "Bundle-Description", "Bundle-DocURL", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-RequiredExecutionEnvironment", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Created-By", "DynamicImport-Package", "Export-Package", "Extension-Name", "Fragment-Host", "Ignore-Package", "Implementation-Build", "Implementation-Build-Date", "Implementation-Title", "Implementation-URL", "Implementation-Vendor", "Implementation-Vendor-Id", "Implementation-Version", "Import-Package", "Include-Resource", "JCabi-Build", "JCabi-Date", "JCabi-Version", "Java-Vendor", "Java-Version", "Main-Class", "Manifest-Version", "Os-Arch", "Os-Name", "Os-Version", "Package", "Private-Package", "Require-Capability", "Scm-Connection", "Scm-Revision", "Scm-Url", "Specification-Title", "Specification-Vendor", "Specification-Version", "Tool", "X-Compile-Source-JDK", "X-Compile-Target-JDK"]
2019-04-18 19:33:39 INFO  Entrypoint:236 - 
[31mOperator[0m has started in version [32m0.6.1[0m.

2019-04-18 19:33:39 INFO  Entrypoint:239 - Git sha: [33mtags/LANG_3_2_1_RC1@r1555582; 2014-01-05 18:01:10+0100[0m
2019-04-18 19:33:39 INFO  Entrypoint:241 - ==================

2019-04-18 19:33:39 INFO  Entrypoint:76 - [33mKubernetes[0m environment detected.
2019-04-18 19:33:43 INFO  AbstractOperator:246 - Starting 'FlinkCluster' operator for namespace flink1
Exception in thread "main" io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:238)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:191)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:384)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:328)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:165)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:607)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:67)
	at io.radanalytics.operator.common.crd.CrdDeployer.initCrds(CrdDeployer.java:34)
	at io.radanalytics.operator.common.AbstractOperator.start(AbstractOperator.java:249)
	at io.radanalytics.operator.Entrypoint.lambda$runForNamespace$6(Entrypoint.java:153)
	at java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:110)
	at java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:557)
	at io.radanalytics.operator.Entrypoint.runForNamespace(Entrypoint.java:136)
	at io.radanalytics.operator.Entrypoint.run(Entrypoint.java:92)
	at io.radanalytics.operator.Entrypoint.main(Entrypoint.java:60)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "nullable" (class io.fabric8.kubernetes.api.model.apiextensions.JSONSchemaProps), not marked as ignorable (36 known properties: "definitions", "default", "$schema", "multipleOf", "minItems", "exclusiveMinimum", "not", "allOf", "additionalItems", "id", "oneOf", "required", "maximum", "exclusiveMaximum", "anyOf", "minProperties", "externalDocs", "dependencies", "example", "$ref", "properties", "format", "additionalProperties", "description", "enum", "minimum", "pattern", "minLength", "patternProperties", "maxLength", "uniqueItems", "maxProperties", "type", "maxItems", "items", "title"])
 at [Source: (BufferedInputStream); line: 1, column: 202859] (through reference chain: io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionList["items"]->java.util.ArrayList[31]->io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinition["spec"]->io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionSpec["validation"]->io.fabric8.kubernetes.api.model.apiextensions.CustomResourceValidation["openAPIV3Schema"]->io.fabric8.kubernetes.api.model.apiextensions.JSONSchemaProps["properties"]->java.util.LinkedHashMap["spec"]->io.fabric8.kubernetes.api.model.apiextensions.JSONSchemaProps["properties"]->java.util.LinkedHashMap["observedConfig"]->io.fabric8.kubernetes.api.model.apiextensions.JSONSchemaProps["nullable"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:823)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1153)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1589)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1567)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:294)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:527)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:364)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:29)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:527)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:364)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:29)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3077)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:236)
	... 15 more

  
#### Expected Results:

  * Normal execution

--- a/pom.xml
+++ b/pom.xml
@@ -2,7 +2,7 @@
     <parent>
         <groupId>io.radanalytics</groupId>
         <artifactId>operator-parent-pom</artifactId>
-        <version>0.3.17</version>
+        <version>0.3.19</version>
     </parent>
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.radanalytics</groupId>

--- a/src/main/java/io/radanalytics/operator/common/crd/CrdDeployer.java
+++ b/src/main/java/io/radanalytics/operator/common/crd/CrdDeployer.java
@@ -1,5 +1,6 @@
 package io.radanalytics.operator.common.crd;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinition;
 import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionBuilder;
 import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionFluent;
@@ -7,6 +8,7 @@ import io.fabric8.kubernetes.api.model.apiextensions.JSONSchemaProps;
 import io.fabric8.kubernetes.client.CustomResourceList;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.utils.Serialization;
 import io.radanalytics.operator.common.EntityInfo;
 import io.radanalytics.operator.common.JSONSchemaReader;
 import org.slf4j.Logger;
@@ -30,6 +32,7 @@ public class CrdDeployer {
         final String newPrefix = prefix.substring(0, prefix.length() - 1);
         CustomResourceDefinition crdToReturn;
 
+        Serialization.jsonMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         List<CustomResourceDefinition> crds = client.customResourceDefinitions()
                 .list()
                 .getItems()

